runtime/lib/registry/Registry.cpp

```diff
@@ -0,0 +1,50 @@
+#include <cstdint>
+#include <cstdio>
+#include <pybind11/pybind11.h>
+#include <unordered_map>
+
+#include "Types.h"
+
+namespace py = pybind11;
+
+// From PyBind11's documentation:
+//
+//     Do you have any global variables that are pybind11 objects or invoke pybind11 functions in
+//     either their constructor or destructor? You are generally not allowed to invoke any Python
+//     function in a global static context. We recommend using lazy initialization and then
+//     intentionally leaking at the end of the program.
+//
+// https://pybind11.readthedocs.io/en/stable/advanced/misc.html#common-sources-of-global-interpreter-lock-errors
+std::unordered_map<int64_t, py::function> *references;
+
+extern "C" {
+[[gnu::visibility("default")]] void callbackCall(int64_t identifier)
```

Comment:
could you remind me why this gnu function attribute is required here?

Reply:
Because pybind11 by default compiles with visibility hidden. This is a good way to override the behaviour as it works in both clang and gcc. The alternative is to make all symbols visible (which I don't think we want) or make a link script for only marking this one externally visible (which maybe we should to at some point, but it would require more time to investigate).

Reply:
thanks!

Reply:
Actually, you may be right here. I think we can remove the visibility here. I think it is possible now that we are using dlopen.

Reply:
Tried it and no. Pybind11 turns all functions into static functions which are not visible and can be inlined.